app/src/main/java/com/example/pygmyhippo/user/EventFragment.java

```diff
@@ -0,0 +1,133 @@
+package com.example.pygmyhippo.user;
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.example.pygmyhippo.R;
+import com.example.pygmyhippo.common.Entrant;
+import com.example.pygmyhippo.common.Event;
+
+import java.util.ArrayList;
+
+/**
+ * This fragment will hold a single events
+ * @author Katharine
+ * @version 1.0
+ * No returns and no parameters
+ */
+public class EventFragment extends Fragment {
+
+    private Event event;
+    // this is the current user who is trying to join the event
+    private Entrant entrant;
+    private ArrayList<Entrant> entrants;
+
+    public EventFragment() {
+    }
+
+    // TODO: to be used for setting text fields in layout
+    public EventFragment(Event event) {
+        this.event = event;
+    }
+
+    // need user information somehow when trying to join this event, user information to be added as an entrant onto the event info list
+    public EventFragment(Event event, Entrant entrant) {
+        this.event = event;
+        this.entrant = entrant;
+    }
+
```

Comment:
you're right, the top two constructors shouldn't ever be used under the assumption that entrant will always exist since the user exists to access the event and the event will always exist given that the user is on the event right now. thanks for catching that!